src/configureWorkspace/configurePython.ts

```diff
@@ -86,8 +96,18 @@ async function genDockerFile(serviceName: string, target: PythonTarget, projectT
         throw new Error(localize('vscode-docker.configurePython.unknownProjectType', 'Unknown project type: {0}', projectType));
     }
 
+    let userSwitchLogic = lowRightsUser;
+    let userWarning = '';
+
+    if (ports?.find(p => p < 1024)) {
+        userSwitchLogic = '';
+        userWarning = rootUserWarning;
```

Comment:
The flip-flopping of the local variables is a little confusing. I'd suggest doing something like:
`const rootPorts: boolean = ports?.some(p => p < 1024) ?? false;`

And below, in the replacements:
`.replace(/\$rootUserWarning\$/g, rootPorts ? rootUserWarning : '')`
and similar for `$user$`